compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaObjectFieldSymbol.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.compiler.api.impl.symbols;
+
+import io.ballerina.compiler.api.impl.SymbolFactory;
+import io.ballerina.compiler.api.symbols.AnnotationSymbol;
+import io.ballerina.compiler.api.symbols.Documentation;
+import io.ballerina.compiler.api.symbols.ObjectFieldSymbol;
+import io.ballerina.compiler.api.symbols.Qualifier;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BAnnotationSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.Symbols;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BField;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.util.Flags;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+import static io.ballerina.compiler.api.symbols.Qualifier.PUBLIC;
+import static io.ballerina.compiler.api.symbols.SymbolKind.OBJECT_FIELD;
+
+/**
+ * Represents a field in an object type descriptor.
+ *
+ * @since 2.0.0
+ */
+public class BallerinaObjectFieldSymbol extends BallerinaSymbol implements ObjectFieldSymbol {
+
+    protected final BField bField;
+    protected List<Qualifier> qualifiers;
+    private final Documentation docAttachment;
+    private final CompilerContext context;
+    private TypeSymbol typeDescriptor;
+    private List<AnnotationSymbol> annots;
+    private String signature;
+    private boolean deprecated;
+
+    public BallerinaObjectFieldSymbol(CompilerContext context, BField bField, SymbolKind kind) {
+        super(bField.name.value, bField.symbol.pkgID, kind, bField.symbol);
+        this.context = context;
+        this.bField = bField;
+        this.docAttachment = new BallerinaDocumentation(bField.symbol.markdownDocumentation);
+        this.deprecated = Symbols.isFlagOn(bField.symbol.flags, Flags.DEPRECATED);
```

Comment:
Suggestion: What if we populate these two fields also on demand as we have done for the other symbols such as annotations, typeDescriptor, etc.

Reply:
+1. Will go ahead and merge this for now. Will send a separate PR for improvements.